extras/indexing/src/main/java/org/apache/rya/api/client/mongo/MongoConnectionDetails.java

```diff
@@ -0,0 +1,119 @@
+    private final String hostname;
+    private final int port;
+
+    /**
+     * Constructs an instance of {@link MongoConnectionDetails}.
+     *
+     * @param hostname - The hostname of the Mongo DB that was connected to. (not null)
+     * @param port - The port of the Mongo DB that was connected to.
+     * @param username - The username that was used to establish the connection
+     *   when performing administrative operations. (not null)
+     * @param password - The password that was used to establish the connection
+     *   when performing administrative operations. (not null)
+     */
+    public MongoConnectionDetails(
+            final String hostname,
+            final int port,
+            final Optional<String> username,
+            final Optional<char[]> password) {
+        this.hostname = requireNonNull(hostname);
+        this.port = port;
+        this.username = requireNonNull(username);
+        this.password = requireNonNull(password);
+    }
+
+    /**
+     * @return The hostname of the Mongo DB that was connected to.
+     */
+    public String getHostname() {
+        return hostname;
+    }
+
+    /**
+     * @return The port of the Mongo DB that was connected to.
+     */
+    public int getPort() {
+        return port;
+    }
+
+     /**
+      * @return The username that was used to establish the connection when performing administrative operations.
+      */
+     public Optional<String> getUsername() {
+         return this.username;
+     }
+
+     /**
+      * @return The password that was used to establish the connection when performing administrative operations.
+      */
+     public Optional<char[]> getPassword() {
+         return password;
+     }
+
+    /**
+     * Create a {@link MongoDBRdfConfiguration} that is using this object's values.
+     *
+     * @param ryaInstanceName - The Rya instance to connect to. (not null)
+     * @return Constructs a new {@link MongoDBRdfConfiguration} object with values from this object.
+     */
+    public MongoDBRdfConfiguration build(final String ryaInstanceName) {
+        // Note, we don't use the MongoDBRdfConfigurationBuilder here because it explicitly sets
+        // authorizations and visibilities to an empty string if they are not set on the builder.
+        // If they are null in the MongoRdfConfiguration object, it may do the right thing.
+        final MongoDBRdfConfiguration conf = new MongoDBRdfConfiguration();
+        conf.setBoolean(ConfigUtils.USE_MONGO, true);
+        conf.setMongoHostname(hostname);
+        conf.setMongoPort("" + port);
+        conf.setMongoDBName(ryaInstanceName);
+
+        if(username.isPresent()) {
+            conf.setMongoUser(username.get());
+        }
+
+        if(password.isPresent()) {
+            conf.setMongoPassword( new String(password.get()) );
+        }
+
+        // Default to having a table prefix of "rya", otherwise the app doesn't work for some reason.
+        conf.setTablePrefix("rya");
```

Comment:
That is strange behavior.  You don't have any idea where it breaks?  Was the table prefix hard coded somewhere else?  Seems like hard coding it here just perpetuates the problem...

Reply:
This requirement is hard coded in RyaSailFactory for unknown reasons, so I'm working around it. It is probably a defect.

Reply:
It looks like there's already a ticket open for this. 
https://issues.apache.org/jira/browse/RYA-135